### ISSUE TITLE
[Issue]: rocprofv2 --kernel-trace doesn't output all dispatches

### ISSUE THREAD

**Issue by guacamoleo** [ROCm/rocprofiler#148]
### Problem Description

rocprofv2 --kernel-trace doesn't output entire trace. I searched for known limitations (seconds of trace data, number of kernels) but couldn't find anything published.

Repro script:
[rocprofv2_issue_repr.txt](https://github.com/user-attachments/files/17544248/rocprofv2_issue_repr.txt)

The script launches 400 kernels but only a fraction of them are recorded in the output, and the number varies with each invocation of rocprofv2.

### Operating System

Ubuntu 22.04.5 LTS (Jammy Jellyfish)

### CPU

AMD EPYC 9454 48-Core Processor

### GPU

AMD Instinct MI300X

### ROCm Version

ROCm 6.2.3

### ROCm Component

rocprofiler

### Steps to Reproduce

1. Rename rocprofv2_issue_repr.txt to rocprofv2_issue_repr.py.
2. rocprofv2 --plugin file --kernel-trace -o output python rocprofv2_issue_repr.py
3. results_output.csv doesn't contain the full 400 kernels dispatched.

### Additional Information

Used docker container based on
FROM rocm/pytorch:rocm6.2.3_ubuntu22.04_py3.10_pytorch_release_2.3.0

**Comment by jrmadsen** [comment]
We are no longer developing rocprofv2. The internal design of rocprofiler v2 was problematic and we came to the conclusion that we needed to scrap it. We re-designed the profiling library from scratch and built a new `rocprofv3` on top of it in ROCm 6.2: [rocprofiler-sdk](https://github.com/ROCm/rocprofiler-sdk). The new profiling library and rocprofv3 are far more stable, reliable, and better tested.
